### ISSUE TITLE
Why not Java 11?

### ISSUE THREAD

**Issue by jsotelo** [linkedin/datahub#1699]
Is there a specific reason we don't use Java 11? Looks like everything runs fine with `./gradlew build` and Java 11.

Java 8 is officially end of life on December 2020.

**Comment by mars-lan** [comment]
Glad to hear that it builds on Java 11 as other users have reported issues building on newer versions of Java. Internally we're still using Java 8 but will most likely move to something newer before EOL.

**Comment by mars-lan** [comment]
By the way, where did you get the EOL date for Java 8? Both [Oracle](https://www.oracle.com/java/technologies/java-se-support-roadmap.html) & [OpenJDK](https://access.redhat.com/articles/1299013) seem to state a much later date than Dec 2020?

**Comment by jsotelo** [comment]
I got it from the table in https://en.wikipedia.org/wiki/Java_version_history but I guess the fine [print](https://en.wikipedia.org/wiki/Java_version_history#cite_note-Java8EOL-1) says "at least".

**Comment by mars-lan** [comment]
Got it. Seems like Oracle has extended the EOL to 2022 now. Regardless, do agree that we should start taking advantage of newer versions of Java. We'll start evaluating the migration internally.

**Comment by jplaisted** [comment]
I want to call this out but we'll need to update our style for new features as well. The big one I can think of is `val` from Java 10, which will probably cause some debate :) I'm on the "avoid at all costs" side of that fence.

**Comment by mars-lan** [comment]
@jplaisted will let you drive this from our side.

**Comment by jplaisted** [comment]
I'll need to dig into this quite a bit more. At the very least we need to provide Java 8 versions until LI and other clients are okay moving to Java 11 for their own deployments. I'm not sure how easy setting up or maintaining the two will be, still need to do more reading.

@jsotelo do you have a specific requirement / need to go to Java 11? Or is it just a nice to have?

==

Good reading

https://medium.com/uptake-tech/migrating-to-java-11-while-maintaining-a-java-8-client-library-f618a3ca6499

**Comment by jsotelo** [comment]
Nothing specific; mainly for better GC [performance](https://optaweb.org/blog/2019/01/17/HowMuchFasterIsJava11.html#:~:text=Java%2011,-Table%201.&text=Almost%20every%20data%20set%20improves,G1%2C%20introduced%20in%20Java%2010) and moving toward the next LTS version. Nothing urgent from my end.

**Comment by jplaisted** [comment]
Thanks! I agree better performance is a win (+ some new profiling tools). Gotta weigh this against the cost of maintaining Java 8 in parallel, for the time being.

**Comment by jjoyce0510** [comment]
@jplaisted Is there updated thinking around this?

**Comment by jplaisted** [comment]
With LI still being on Java 8 I haven't had time to look at this. Though we're going to Java 11 in theory by the end of this quarter... I can revisit this then, unless you want to carry this forward before then. In the meantime we'd still at least need Java 8 compatible GMA libraries.

**Comment by maggiehays** [comment]
We are now managing feature requests through our [new portal](https://feature-requests.datahubproject.io) to improve visibility & prioritize work into our public roadmap. If this request is still relevant, please create a new issue so we can gauge current interest from the Community & prioritize work accordingly.

We'll close out this issue in the meantime. Feel free to [reach out to me on DataHub Slack](https://slack.datahubproject.io/) with any questions!

**Comment by rakhiagr** [comment]
We are planning to update to Java 11 this quarter. What are your views? @jjoyce0510 @jsotelo
